features/telemetry/daemon/src/main/java/org/opennms/netmgt/telemetry/daemon/KarafStartupMonitor.java

```diff
@@ -0,0 +1,135 @@
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.daemon;
+
+import java.util.Base64;
+
+import org.opennms.core.soa.lookup.ServiceLookup;
+import org.opennms.core.soa.lookup.ServiceLookupBuilder;
+import org.opennms.core.soa.lookup.ServiceRegistryLookup;
+import org.opennms.core.soa.support.DefaultServiceRegistry;
+import org.opennms.netmgt.daemon.DaemonTools;
+import org.opennms.netmgt.daemon.SpringServiceDaemon;
+import org.opennms.netmgt.events.api.EventConstants;
+import org.opennms.netmgt.events.api.annotations.EventHandler;
+import org.opennms.netmgt.events.api.annotations.EventListener;
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.reporting.core.svclayer.ReportStoreService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Used in startup script to detect if Karaf started properly.
+ * TODO: patrick move this class to better place
+ */
+@EventListener(name= KarafStartupMonitor.NAME, logPrefix= KarafStartupMonitor.LOG_PREFIX)
+public class KarafStartupMonitor implements SpringServiceDaemon {
+    private static final Logger LOG = LoggerFactory.getLogger(KarafStartupMonitor.class);
+
+    public static final String NAME = "KarafStartupMonitor";
+
+    public static final String LOG_PREFIX = "karafStartupMonitor";
+
+    @SuppressWarnings("unchecked")
+    private static final ServiceLookup<Class<?>, String> SERVICE_LOOKUP = new ServiceLookupBuilder(new ServiceRegistryLookup(DefaultServiceRegistry.INSTANCE))
+            .blocking()
+            .build();
+
+    @Override
+    public synchronized void start() {
+        LOG.info("{} is starting.", NAME);
+
+        if (!isKarafOk()) {
+        String message = String.format(NAME + ": It seems Kafka can't be started properly. This is bad, will fail startup.%n" +
+        "What can you do about this?%n" +
+        "1.) check in logs/karaf.log for problems%n" +
+        "2.) clear the 'data' folder - it contains Karaf's cache%n" +
+        "3.) run the script bin/fix-karaf-setup.sh");
+            throw new IllegalStateException(message);
+        }
+        LOG.info("{} is started.", NAME);
+    }
+
+    private boolean isKarafOk() {
+        Class serviceClass = ReportStoreService.class;
+        try {
+            final ReportStoreService service = SERVICE_LOOKUP.lookup(serviceClass, null);
+            return service != null;
+        } catch (Exception e) {
+            LOG.warn("Could not lookup {}, will fail.", serviceClass.getName(), e);
+            return false;
+        }
+    }
+
+    private boolean isKarafOkAlternative() {
```

Comment:
this seems to work as well as an alternative but isKarafOk() should do => we can delete this method